src/System.Data.SqlClient/src/System/Data/Common/AdapterUtil.cs

```diff
@@ -1142,5 +1142,21 @@ internal static bool IsAzureSqlServerEndpoint(string dataSource)
 
             return false;
         }
+
+        static internal ArgumentOutOfRangeException InvalidDataRowVersion(DataRowVersion value)
+        {
+#if DEBUG
```

Comment:
I tried this out. 
The IL is generated for Switch Case without the #if DEBUG as well with empty body
```
// System.Data.Common.ADP
internal static ArgumentOutOfRangeException InvalidDataRowVersion(DataRowVersion value)
{
	if (value <= 512)
	{
		if (value != 256 && value != 512)
		{
		}
	}
	else if (value != 1024)
	{
	}
	return ADP.InvalidEnumerationValue(typeof(DataRowVersion), value);
}
```